subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -9,6 +9,9 @@
  * @brief WiFi shell module
```

Comment:
I would prefer that the keep the source code of the shell commands in wifi_credentials directory and the file. The wifi_shell.c is quite large already so don't want to clutter it with extra stuff. Sorry that I did not manage to explain myself clearly.
So I did similar thing with the bridge shell, and moved the shell **functionality** under the net shell but left the code in where it is now. See this commit https://github.com/zephyrproject-rtos/zephyr/pull/77235/commits/54cd26e1a162cb2719a9d5c5df6fa4e5afa457cd for an example.
Similar thing could be done with wifi credentials.

Reply:
@jukkar @rlubos I tried to do that. It requires a bigger change. What do you think?

Reply:
I don't think the change is that big. The `wifi_commands` in `wifi_shell.c` needs to be split into individual pieces, meaning for each subcommand a `SHELL_SUBCMD_ADD()` needs to created.
I did an experiment and got following diff, it converted the "wifi ap" command, but I did not touch the others in wifi_commands.
```
diff --git a/subsys/net/l2/wifi/wifi_shell.c b/subsys/net/l2/wifi/wifi_shell.c
index 64bee4035ae..9d3df0b7161 100644
--- a/subsys/net/l2/wifi/wifi_shell.c
+++ b/subsys/net/l2/wifi/wifi_shell.c
@@ -2692,6 +2692,17 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_cmd_ap,
 	SHELL_SUBCMD_SET_END
 );
 
+static int cmd_wifi_ap(const struct shell *sh, size_t argc, char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	return 0;
+}
+
+SHELL_SUBCMD_ADD((wifi), ap, &wifi_cmd_ap,
+		 "Access Point mode commands.", cmd_wifi_ap, 0, 0);
+
 SHELL_STATIC_SUBCMD_SET_CREATE(wifi_twt_ops,
 	SHELL_CMD_ARG(quick_setup, NULL, " Start a TWT flow with defaults:\n"
 		"<twt_wake_interval: 1-262144us> <twt_interval: 1us-2^31us>.\n",
@@ -2798,7 +2809,6 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 	SHELL_CMD_ARG(version, NULL, "Print Wi-Fi Driver and Firmware versions\n",
 		  cmd_wifi_version,
 		  1, 0),
-	SHELL_CMD(ap, &wifi_cmd_ap, "Access Point mode commands.\n", NULL),
 	SHELL_CMD_ARG(connect, NULL,
 		  "Connect to a Wi-Fi AP\n"
 		  "<-s --ssid \"<SSID>\">: SSID.\n"
@@ -2936,7 +2946,9 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 	SHELL_SUBCMD_SET_END
 );
 
-SHELL_CMD_REGISTER(wifi, &wifi_commands, "Wi-Fi commands", NULL);
+SHELL_SUBCMD_SET_CREATE(wifi_cmds, (wifi));
+
+SHELL_CMD_REGISTER(wifi, &wifi_cmds, "Wi-Fi commands", NULL);
 
 static int wifi_shell_init(void)
 {
diff --git a/subsys/net/lib/wifi_credentials/wifi_credentials_shell.c b/subsys/net/lib/wifi_credentials/wifi_credentials_shell.c
index ef025982326..72c78281299 100644
--- a/subsys/net/lib/wifi_credentials/wifi_credentials_shell.c
+++ b/subsys/net/lib/wifi_credentials/wifi_credentials_shell.c
@@ -289,18 +289,17 @@ static int cmd_list_networks(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
-SHELL_SUBCMD_SET_CREATE(sub_wifi_cred, (wifi_cred));
-SHELL_SUBCMD_ADD((wifi_cred), add, NULL, "Add network to storage.", cmd_add_network, 0, 0);
-SHELL_SUBCMD_ADD((wifi_cred), delete, NULL, "Delete network from storage.", cmd_delete_network, 0,
-		 0);
-SHELL_SUBCMD_ADD((wifi_cred), list, NULL, "List stored networks.", cmd_list_networks, 0, 0);
-
-SHELL_CMD_REGISTER(wifi_cred, &sub_wifi_cred, "Wi-Fi Credentials commands", NULL);
-
-#if CONFIG_WIFI_CREDENTIALS_CONNECT_STORED
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_wifi_cred,
+       SHELL_CMD(add, NULL, "Add network to storage.", cmd_add_network),
+       SHELL_CMD(delete, NULL, "Delete network from storage.", cmd_delete_network),
+       SHELL_CMD(list, NULL, "List stored networks.", cmd_list_networks),
+       SHELL_SUBCMD_SET_END
+);
 
 static int cmd_auto_connect(const struct shell *sh, size_t argc, char *argv[])
 {
+#if CONFIG_WIFI_CREDENTIALS_CONNECT_STORED
+
 	struct net_if *iface = net_if_get_first_by_type(&NET_L2_GET_NAME(ETHERNET));
 	int rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, iface, NULL, 0);
 
@@ -309,11 +308,10 @@ static int cmd_auto_connect(const struct shell *sh, size_t argc, char *argv[])
 			    "An error occurred when trying to auto-connect to a network. err: %d",
 			    rc);
 	}
-
-	return 0;
-}
-
-SHELL_SUBCMD_ADD((wifi_cred), auto_connect, NULL, "Connect to any stored network.",
-		 cmd_auto_connect, 0, 0);
-
 #endif /* CONFIG_WIFI_CREDENTIALS_CONNECT_STORED */
+
+	return 0;
+}
+
+SHELL_SUBCMD_ADD((wifi), cred, &sub_wifi_cred, "Connect to any stored network.",
+		 cmd_auto_connect, 0, 0);
```
I am not sure how the `auto_connect` is suppose to work, but just did a quick hack to get things working.

I am pushing this change as I would like the wifi_shell.c to be split into smaller pieces like what we did for net-shell. So all this work is preparation for that. It is causing problems if we have "wifi_cred" shell and then later move it to wifi shell and need to change the documentation etc.